fairlearn/datasets/_synthetic.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+from collections import OrderedDict
+from itertools import product
+
+import numpy as np
+import pandas as pd
+from sklearn.datasets import make_classification
+from sklearn.utils import check_random_state
+
+
+class SensitiveFeatureGroupConfig:
+
+    def __init__(self, group_dict, **classification_kwargs):
+        """ `group_dict` is a dict assigning to a group in each sensitive
+        feature category. E.g.,::
+            {
+                'gender': 'Other',
+                'age': '50-60',
+            }
+        `classification_kwargs` is the set of kwargs that will be passed on to
+        make_classification.
+        """
+        self.group_dict = group_dict
+        self.classification_kwargs = classification_kwargs
+
+    def __repr__(self):
+        return f'{self.group_dict}={self.classification_kwargs}'
+
+
+class SensitiveFeature:
+    """
+    Parameters
+    ----------
+    name : str
+        Name of the feature
+
+    groups : list of str
+        List of groups for the feature
+    """
+
+    def __init__(self, name, groups):
+        self.name = name
+        self.groups = groups
+
+    def __repr__(self):
+        return f'{self.name}[{",".join(self.groups)}]'
+
+
+class SensitiveDatasetMaker:
+
+    def __init__(self, sensitive_features=None, random_state=None):
+        self.rng = check_random_state(random_state)
+        self.sensitive_features = sensitive_features or []
```

Comment:
I'm surprised `sensitive_features` is optional. What's the use case for having `sensitive_features=[]`?

Reply:
The idea was that one could instantiate the `SensitiveDatasetMaker` and add sensitive features later, then manually call `init_configured_groups`, but I think this is too complicated. I've removed this option.